### PR TITLE
Feat/filter sort search

### PR DESCRIPTION
**Pull request by Aditi0505**
# Change Logs:

## Description:

- Added Filter by difficulty level
- Added Sort functionality in alphabetical order
- Added Search functionality on the Quiz Category display page

## Screenshot 
image
image

## Preview Link:
https://deploy-preview-7--guessthisquiz.netlify.app/

### CODE REVIEW

File: src/components/Radiobutton/Radiobutton.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { useFilter } from "../../context";
+const Radiobutton = ({ value, name, type, payload }) => {
+  const { filterState, filterDispatch } = useFilter();
+  console.log(payload, filterState.sortBy);
```
Comment (rohanmathur91) [inline src/components/Radiobutton/Radiobutton.jsx:4]: remove console log

Comment (rohanmathur91) [inline src/components/Radiobutton/Radiobutton.jsx:4]: remove console log

**Review by rohanmathur91 — APPROVED** [review]
- Good folder structure and clean coding
- Nice naming for components and good use of utility functions
- Overall looks good, great work!

File: src/index.js [hunk 2]
```diff
@@ -5,20 +5,23 @@ import App from "./App";
...
-        <QuestionProvider>
-          <ThemeProvider>
-            <App />
-          </ThemeProvider>
-        </QuestionProvider>
-      </CurrentQuestionProvider>
+      <FilterProvider>
+        <CurrentQuestionProvider>
+          <QuestionProvider>
+            <ThemeProvider>
```
Comment (Rohitprasad83) [inline src/index.js:19]: put all of these providers into one file

File: src/reducer/currentQuestion-reducer.js [hunk 3]
```diff
@@ -11,6 +11,15 @@ const currentQuestionReducer = (state, action) => {
           action.payload.selectedOption === action.payload.correctAnswer
             ? state.score + 10
             : state.score,
+        results: [
+          ...state.results,
+          {
+            currentQuestion: action.payload.currentQuestion,
+            questions: action.payload.question,
```
Comment (Rohitprasad83) [inline src/reducer/currentQuestion-reducer.js:18]: can destructure the action.payload

**Review by Rohitprasad83 — APPROVED** [review]
- Code is Readable
- Logic Implementation is Good
- just added a couple of suggestions

File: src/index.js (hunk 2 above)
Comment (Aditi0505) [reply src/index.js:19]: I will implement this later on.

**Pull request merged**